lsp-rust.el

```diff
@@ -484,6 +489,32 @@ PARAMS progress report notification data."
                                    (when (lsp-find-workspace 'rust-analyzer nil)
                                      (lsp-rust-analyzer-inlay-hints-mode)))))
 
+(defun lsp-rust-analyzer-expand-macro ()
+  "Expands the macro call at point recursively."
+  (interactive)
+  (-if-let ((workspace (lsp-find-workspace 'rust-analyzer default-directory)))
+      (-if-let* ((params (list :textDocument (lsp--text-document-identifier)
+                               :position (lsp--cur-position)))
+                 (response (with-lsp-workspace workspace
+                             (lsp-send-request (lsp-make-request
```

Comment:
nit: you may use lsp-request and lsp--text-document-position-params

Reply:
I will make these changes for the whole file in a new pull request.